### ISSUE TITLE
Doesn't support arm64 (aarch64) instruction set

### ISSUE THREAD

**Issue by vegabook** [matrix-org/synapse#5685]
**Description:**

Please could you support Arm64 instruction set in your repo. Jetson family of devices from Nvidia is very popular, especially since the Nano was launched. Also RPI4, now with 4GB, is already wildly successful with all outlets out of stock, and a bunch of Linux distros have announced they will target its 64 bit capabilities. 

Output:

```
tbrowne@nano:~$ wget -qO - https://matrix.org/packages/debian/repo-key.asc | sudo apt-key add -
OK
tbrowne@nano:~$ sudo add-apt-repository https://matrix.org/packages/debian/
Get:1 file:/var/cuda-repo-10-0-local-10.0.166  InRelease
Ign:1 file:/var/cuda-repo-10-0-local-10.0.166  InRelease
Get:2 file:/var/visionworks-repo  InRelease
Ign:2 file:/var/visionworks-repo  InRelease
Get:3 file:/var/visionworks-sfm-repo  InRelease
Ign:3 file:/var/visionworks-sfm-repo  InRelease
Get:4 file:/var/visionworks-tracking-repo  InRelease                     
Ign:4 file:/var/visionworks-tracking-repo  InRelease                     
Get:5 file:/var/cuda-repo-10-0-local-10.0.166  Release [574 B]           
Hit:6 http://ports.ubuntu.com/ubuntu-ports bionic InRelease                                       
Hit:7 http://archive.canonical.com bionic InRelease                      
Get:5 file:/var/cuda-repo-10-0-local-10.0.166  Release [574 B]           
Hit:8 http://ports.ubuntu.com/ubuntu-ports bionic-updates InRelease
Hit:9 http://ports.ubuntu.com/ubuntu-ports bionic-backports InRelease
Hit:10 http://ports.ubuntu.com/ubuntu-ports bionic-security InRelease
Get:11 file:/var/visionworks-repo  Release [1,999 B]              
Get:11 file:/var/visionworks-repo  Release [1,999 B]              
Get:12 file:/var/visionworks-sfm-repo  Release [2,003 B]              
Get:12 file:/var/visionworks-sfm-repo  Release [2,003 B]              
Get:13 file:/var/visionworks-tracking-repo  Release [2,008 B]         
Get:13 file:/var/visionworks-tracking-repo  Release [2,008 B]         
Get:14 https://packages.matrix.org/debian bionic InRelease [2,906 B]  
Fetched 2,906 B in 3s (991 B/s)
Reading package lists... Done
N: Skipping acquire of configured file 'main/binary-arm64/Packages' as repository 'https://matrix.org/packages/debian bionic InRelease' doesn't support architecture 'arm64'
```

**Comment by vegabook** [comment]
I have been able to solve this by commenting out the non-src line and  uncommending deb-src

```
# deb https://matrix.org/packages/debian/ bionic main
deb-src https://matrix.org/packages/debian/ bionic main

```

**Comment by Ra5pi** [comment]
> I have been able to solve this by commenting out the non-src line and uncommending deb-src
> 
> ```
> # deb https://matrix.org/packages/debian/ bionic main
> deb-src https://matrix.org/packages/debian/ bionic main
> ```

Hey, sounds like you managed to install it on a raspberry?
Could you share the exact steps?
I am still struggling installing it on a rpi 3
Thanks!

**Comment by vanjavk** [comment]
> > I have been able to solve this by commenting out the non-src line and uncommending deb-src
> > ```
> > # deb https://matrix.org/packages/debian/ bionic main
> > deb-src https://matrix.org/packages/debian/ bionic main
> > ```
> 
> Hey, sounds like you managed to install it on a raspberry? Could you share the exact steps? I am still struggling installing it on a rpi 3 Thanks!

same

**Comment by vegabook** [comment]
> > I have been able to solve this by commenting out the non-src line and uncommending deb-src
> > ```
> > # deb https://matrix.org/packages/debian/ bionic main
> > deb-src https://matrix.org/packages/debian/ bionic main
> > ```
> 
> Hey, sounds like you managed to install it on a raspberry? Could you share the exact steps? I am still struggling installing it on a rpi 3 Thanks!

Hey, at the time this referred to the `/etc/apt/sources.list.d` file that [synapse installation instructions](https://matrix-org.github.io/synapse/latest/setup/installation.html#debianubuntu) created. So you needed to change to the _source_ rather than precompiled, and then it would compile from source for you. 

Just tried that again now and seems to be an error. I'm afraid I'm not going to debug until Ubuntu 22.04 comes out in a few weeks - will update this post then but if you want to give it a shot along the above lines that's the way to go. 

Basically - I needed to compile from source and that _used to_ be achievable with the above change to the apt file but seems like some more tweaking may be necessary.
